src/org/zaproxy/zap/extension/websocket/ExtensionWebSocket.java

```diff
@@ -433,6 +440,28 @@ public void removeAllChannelObserver(WebSocketObserver observer) {
 		}
 		allChannelObservers.remove(observer);
 	}
+	
+	/**
+	 * Add an sender listener that is attached to every channel connected in future.
+	 * 
+	 * @param senderListener
+	 */
+	public void addAllChannelSenderListener(WebSocketSenderListener senderListener){
+		allChannelSenderListeners.add(senderListener);
+	}
+	
+	/**
+	 * Removes the given {@code senderListener}, that was attached to every channel connected.
+	 * 
+	 * @param senderListener the sender listener to be removed
+	 * @throws IllegalArgumentException if the given {@code senderListener} is {@code null}.
+	 */
+	public void removeAllChannelSenderListener(WebSocketSenderListener senderListener){
+		if (senderListener == null) {
+			throw new IllegalArgumentException("The parameter senderListener must not be null.");
+		}
+		allChannelSenderListeners.remove(senderListener);
```

Comment:
Exactly. `removeAllChannelObserver` seems to have a same bug. 
https://github.com/zaproxy/zap-extensions/blob/master/src/org/zaproxy/zap/extension/websocket/ExtensionWebSocket.java#L430-L435

May I fix both?